android/cli/commands/_build.js

```diff
@@ -1472,12 +1472,24 @@ AndroidBuilder.prototype.validate = function validate(logger, config, cli) {
 				if (module.platform.indexOf('commonjs') != -1) {
 					module.native = false;
 
+					// Look for legacy module.id.js first
 					module.libFile = path.join(module.modulePath, module.id + '.js');
 					if (!fs.existsSync(module.libFile)) {
-						this.logger.error(__('Module %s version %s is missing module file: %s', module.id.cyan, (module.manifest.version || 'latest').cyan, module.libFile.cyan) + '\n');
-						process.exit(1);
+						// then package.json TODO Verify the main property points at reale file under the module!
+						module.libFile = path.join(module.modulePath, 'package.json');
```

Comment:
You must not use `module.libFile` to store the path as you run your checks. Use a local variable and when you have a match, then set `module.libFile`.

Currently, if the `<module id>.js` file doesn't exist, then you proceed to check for a `package.json`, which is fine. However, if the module id file doesn't exist and there's no `package.json`, then the libFile will be the `package.json` which is very wrong. `package.json` should never be the "libFile".

Anyways, read the `package.json` and check for a `main` property, then check if the `main` file exists. If the `main` file doesn't exist, then continue on to the `index.js`.

Having a `.json` file as the main module file should not be supported by Titanium. The CommonJS spec does not describe `.json` files as being valid modules. Please remove the `index.json` checks.